src/posts/posts.service.ts

```diff
@@ -50,18 +52,22 @@ export class PostsService {
 
   async createPost(
     postCreationDto: PostCreationDto,
-    userId: number,
+    userId: string,
   ): Promise<PostCreationInterface> {
+    // get user to add post to it
+    const user = await this.userRepository.findOne({ where: { uuid: userId } });
+
+    // create the post
     const createdPost = await this.postRepository.createPost(
       postCreationDto,
-      userId,
+      user,
     );
     return { id: createdPost.uuid };
   }
 
-  async flagPost(postId: string, flag: boolean, userId: number): Promise<void> {
+  async flagPost(postId: string, flag: boolean, userId: string): Promise<void> {
     // get post
-    const post = await this.postRepository.getPostById(postId);
+    const post = await this.postRepository.getPostWithUserById(postId);
```

Comment:
there's no need to Change the method name

Reply:
I find it more explicit to do so. Also to be consistent with other methods names'

Reply:
you still getPostById.. doesn't make sense to keep changing method name when you add extra logic or parameters. if you are getting the post with something else other than user, would you change the method name then?I still see it as getPostById

Reply:
done